docs/BEST_PRACTICES.md

```diff
@@ -6,10 +6,17 @@ This is a collection of tips, advice, gotchas and other best practices for using
 * Do all real work that involves running a shell script, building, etc, within `node` blocks, so that it actually happens on a real executor, rather than a flyweight executor on the master node.
 * Get your flows from source control - `Jenkinsfile`s, loading libraries, global [CPS](https://en.wikipedia.org/wiki/Continuation-passing_style) library, you name it - but if you pull the main flow from SCM (i.e., Multibranch with `Jenkinsfile`s or `Pipeline from SCM`), be aware that you may need to whitelist a lot of method calls in the script security plugin. By getting your flows from source control, you benefit from `Jenkinsfile` versioning and also testing and merging against your CD Pipeline definition.
 * `input` shouldn’t be done within a `node` block. For `input` step is recommended to use `timeout` in order to avoid waiting for an infinite amount of time, and also control structures (`try/catch/finally`).
-* As Pipeline usage is adopted for multiple projects and teams in an organization, common patterns should be stored in [Shared Libraries](https://jenkins.io/doc/book/pipeline/shared-libraries/). It is also an escape value for allowing out-of-sandbox execution in a safe context.
 * When writing functions, use unique names in your pipeline script and avoid using built-in/pre-defined items (such as "build", "stage", etc).  Using pre-defined methods may result in runtime issues, such as generating a `sandbox.RejectedAccessException` error when using build job DSL. 
 * Make use of the available [Pipeline Development Tools](https://jenkins.io/doc/book/pipeline/development/#pipeline-development-tools) for debugging your Pipeline as code.
 * Use [Multibranch Pipeline](https://jenkins.io/doc/book/pipeline/multibranch/) for project collaboration, new features (developed in separate branches) are validated before merging them to the master branch. Besides, it comes with automating features out-of-the-box (webhooks).
+
+# Shared Libraries
+* As Pipeline usage is adopted for multiple projects and teams in an organization, common patterns should be stored in [Shared Libraries](https://jenkins.io/doc/book/pipeline/shared-libraries/). It is also an escape value for allowing out-of-sandbox execution in a safe context.
+* Pipelines [Shared Libraries can be integrated with Third Party libraries](https://jenkins.io/doc/book/pipeline/shared-libraries/#using-third-party-libraries) however the best approach is to place that integration into an **external process** (groovy or any other languages) run by `sh`/`bat` steps, for instance:
```

Comment:
Unclear what you're saying.  I think better phased as: 

"The best way to implement complex integrations or use Third-Party libraries is external scripts.  These may be written in Groovy, Python, or other languages and are easy to invoke with `sh`/`bat` steps: (syntax)" 

Make sure to include one example with 'returnStdout: true' i.e. `def output = sh 'groovy myScript.groovy $args'  \\ Lets you capture the output from the script'`

Then add a separate bullet point mentioning that Pipeline *can* load Third-Party Groovy libraries (with link) - but this feature should be used with caution.  The libraries will be executed on-master, so they carry additional load, and run into conflicts with Pipeline internal logic or difficult-to-resolve dependency issues.